File: C++/Easy/WrongAnswer/distribute-money-to-maximum-children_0.cpp
class Solution {
public:
    int distMoney(int money, int children) {
        
        if (money < children + 1) return -1;
        if (money == children + 1) return children - 1;
        if (money == 2 * children + 1) return children;
        if (money == 3 * children + 1) return children + 1;
        if (money == 4 * children + 1) return -1;
        if (money >= 5 * children + 1) return children;
        return -1;
    }
};

File: Rust/Easy/WrongAnswer/distribute-money-to-maximum-children_0.rs
impl Solution {
    pub fn dist_money(money: i32, children: i32) -> i32 {
        
        let mut res = -1;
        for i in 1..children {
            if money - (i * 8) >= (children - i) {
                res = i;
            }
        }
        res
    }
}